plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsRepository.java

```diff
@@ -142,7 +144,14 @@ private HdfsBlobStore createBlobstore(URI blobstoreUri, String path, Settings re
         );
 
         try {
-            return new HdfsBlobStore(fileContext, path, bufferSize, isReadOnly(), haEnabled);
+            return new HdfsBlobStore(
+                fileContext,
+                path,
+                bufferSize,
+                isReadOnly(),
+                haEnabled,
+                replicationFactor != null ? replicationFactor.shortValue() : null
```

Comment:
Good point! I've added validation against `dfs.replication.min` and `dfs.replication.max`.

I've had to make the validation in `createBlobstore`, since that's where `hadoopConfiguration` is first created.
That makes the outside exception say `path is not accessible` when validation error occurs, which is a bit confusing:

```
{
  "error": {
    "root_cause": [
      {
        "type": "repository_exception",
        "reason": "[estest] Value of create_opts.replication [-2] must be >= dfs.replication.min [1]"
      }
    ],
    "type": "repository_verification_exception",
    "reason": "[estest] path  is not accessible on master node",
    "caused_by": {
      "type": "repository_exception",
      "reason": "[estest] Value of create_opts.replication [-2] must be >= dfs.replication.min [1]"
    }
  },
  "status": 500
}
```

Though it's also like that when you specify a wrong HDFS address, so I suppose that could be OK?

Alternatively, I could just hard-code a check for a 1 to 32767 range instead.

Reply:
Hmm, that error message seems to be tied to the verification logic in the `BlobStoreRepository` class. If it doesn't have a version of the repository yet when doing the verification, it creates it, but the creation is done inside of the try catch block that reports an error about not being able to write to a path. I think as long as you're doing the validation inside of the `createBlobstore` method, that is the right place. The ultimate error message is likely out of scope for this PR.